src/py/flwr/client/supernode/app.py

```diff
@@ -77,9 +74,15 @@ def run_supernode() -> None:
         )
         sys.exit(1)
 
+    # If neither `--insecure` nor `--root-certificates` are set, exit
+    if not (args.insecure) and args.root_certificates is None:
+        log(
+            ERROR,
+            "'--root-certificates' are required if not running with '--insecure' mode.",
+        )
+        return
+
```

Comment:
Should we move this to `try_obtain_root_certificates`?

Reply:
Resolved in #4541.